content/enterprise_influxdb/v1.9/flux/optimize-queries.md

```diff
@@ -0,0 +1,182 @@
+
+#### Pushdown functions and function combinations
+The following pushdowns are supported in InfluxDB Enterprise 1.9+.
+
+| Functions                      | Supported            |
+| :----------------------------- | :------------------: |
+| **count()**                    | {{< icon "check" >}} |
+| **drop()**                     | {{< icon "check" >}} |
+| **duplicate()**                | {{< icon "check" >}} |
+| **filter()** {{% req " \*" %}} | {{< icon "check" >}} |
+| **fill()**                     | {{< icon "check" >}} |
+| **first()**                    | {{< icon "check" >}} |
+| **group()**                    | {{< icon "check" >}} |
+| **keep()**                     | {{< icon "check" >}} |
+| **last()**                     | {{< icon "check" >}} |
+| **max()**                      | {{< icon "check" >}} |
+| **mean()**                     | {{< icon "check" >}} |
+| **min()**                      | {{< icon "check" >}} |
+| **range()**                    | {{< icon "check" >}} |
+| **rename()**                   | {{< icon "check" >}} |
+| **sum()**                      | {{< icon "check" >}} |
+| **window()**                   | {{< icon "check" >}} |
+| _Function combinations_        |                      |
+| **window()** \|> **count()**   | {{< icon "check" >}} |
+| **window()** \|> **first()**   | {{< icon "check" >}} |
+| **window()** \|> **last()**    | {{< icon "check" >}} |
+| **window()** \|> **max()**     | {{< icon "check" >}} |
+| **window()** \|> **min()**     | {{< icon "check" >}} |
+| **window()** \|> **sum()**     | {{< icon "check" >}} |
+
+{{% caption %}}
+{{< req "\*" >}} **filter()** only pushes down when all parameter values are static.
+See [Avoid processing filters inline](#avoid-processing-filters-inline).
+{{% /caption %}}
+
+Use pushdown functions and function combinations at the beginning of your query.
+Once a non-pushdown function runs, Flux pulls data into memory and runs all
+subsequent operations there.
+
+##### Pushdown functions in use
+```js
+from(bucket: "db/rp")
+  |> range(start: -1h)                       //
+  |> filter(fn: (r) => r.sensor == "abc123") //
+  |> group(columns: ["_field", "host"])      // Pushed to the data source
+  |> aggregateWindow(every: 5m, fn: max)     //
+  |> filter(fn: (r) => r._value >= 90.0)     //
+
+  |> top(n: 10)                              // Run in memory
+```
+
+### Avoid processing filters inline
+Avoid using mathematic operations or string manipulation inline to define data filters.
+Processing filter values inline prevents `filter()` from pushing its operation down
+to the underlying data source, so data returned by the
+previous function loads into memory.
+This often results in a significant performance hit.
+
+For example, the following query uses [dashboard variables](/influxdb/v2.0/visualize-data/variables/)
```

Comment:
update link to -> /enterprise_influxdb/v1.9/ ??

Reply:
Do these exist in 1.x?

Reply:
I'll update the example to not use dashboard variables. In 1.x land, these would be Chronograf template variables.